libmscore/tie.cpp

```diff
@@ -563,31 +574,46 @@ void Tie::calculateDirection()
                   //
                   // chords
                   //
-                  QList<int> ties;
-                  int idx = 0;
-                  int noteIdx = -1;
+                  int tiesCount = 0;
+                  Note* tieNote = startNote();
+                  Note* tieAbove = nullptr;
+                  Note* tieBelow = nullptr;
+
                   for (size_t i = 0; i < n; ++i) {
                         if (notes[i]->tieFor()) {
-                              ties.append(notes[i]->line());
-                              if (notes[i] == startNote()) {
-                                    idx = ties.size() - 1;
-                                    noteIdx = int(i);
+                              tiesCount++;
+                              int noteDiff = compareNotesPos(notes[i], tieNote);
+
+                              if (noteDiff > 0) {
+                                    if (!tieAbove)
+                                          tieAbove = notes[i];
+                                    else if (compareNotesPos(notes[i], tieAbove) < 0)
+                                          tieAbove = notes[i];
+                                    else if (noteDiff < 0) {
```

Comment:
`if (noteDiff > 0)` should close before the `else if (noteDiff < 0)` block starts

Reply:
I might have removed a couple braces too much I guess?